File: 103-BinaryTreeZigzagLevelOrderTraversal/103-BinaryTreeZigzagLevelOrderTraversal.py
# Last updated: 07/04/2026, 11:41:35
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def zigzagLevelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if root is None :
            return 
        ans = []
        
        que = [root]
        m=-1
        while que :
            
            m+=1
            k = len(que)

            s = []
            for i in range(k):
                j = que.pop(0)
                s.append(j.val)
                if j.left :
                    que.append(j.left)
                if j.right :
                    que .append(j.right)

            if m %2 == 0 :
                ans.append(s)
            else:

                ans.append(s[::-1])

        return ans

File: 103-BinaryTreeZigzagLevelOrderTraversal/103-BinaryTreeZigzagLevelOrderTraversal.cpp
// Last updated: 07/04/2026, 11:41:30
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> zigzagLevelOrder(TreeNode* root) {
        
        vector<vector<int>> ans;
        queue<TreeNode *> st;
        if ( root==nullptr) return ans;

        st.push(root);
        
        int m = -1;
        
        while(!(st.empty())){
            
            m++;
            int k = st.size();
            vector<int> temp;

            for ( int i = 0;i<k;i++){
                
                root = st.front();
                st.pop();
                temp.push_back(root->val);
                if (root->left) st.push(root->left);
                if (root->right) st.push(root->right);
 
            }

            if ( m%2 ==1){
                reverse(temp.begin(),temp.end());
            }
            ans.push_back(temp);

            
        }
        return ans;







    }
};